From 3ca28812b411c9db9c1c3300639872e84f490796 Mon Sep 17 00:00:00 2001
From: Anu Engineer <aengineer@apache.org>
Date: Fri, 17 Jun 2016 23:29:57 -0700
Subject: [PATCH 1620/2748] HDFS-10541. Diskbalancer: When no actions in plan,
 error message says "Plan was generated more than
 24 hours ago". Contributed by Anu Engineer.

(cherry picked from commit 6f301d56015ba1a21bf33feeb00b4ea9f016ea6c)

Change-Id: I40eda4a4a0774fe7a01c47edc46c2ea548170a9b
---
 .../server/diskbalancer/command/PlanCommand.java   |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/PlanCommand.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/PlanCommand.java
index 91ab7fb..4e2febb 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/PlanCommand.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/diskbalancer/command/PlanCommand.java
@@ -137,7 +137,7 @@ public void execute(CommandLine cmd) throws Exception {
           .getBytes(StandardCharsets.UTF_8));
     }
 
-    if (plan != null) {
+    if (plan != null && plan.getVolumeSetPlans().size() > 0) {
       LOG.info("Writing plan to : {}", getOutputPath());
       try (FSDataOutputStream planStream = create(String.format(
           DiskBalancer.PLAN_TEMPLATE,
-- 
1.7.9.5

